contrib/automation_tests/fragments/connection_window.py

```diff
@@ -0,0 +1,81 @@
+"""
+Copyright (c) 2020 The Orbit Authors. All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file.
+"""
+
+import logging
+import time
+
+from absl import flags
+
+from core.orbit_e2e import E2ETestCase, wait_for_condition
+
+
+flags.DEFINE_bool('enable_ui_beta', False, 'Expect Orbit to be started with the new UI')
+
+
+class ConnectToStadiaInstance(E2ETestCase):
+    """
+    Connect to the first available stadia instance
+    """
+    def _execute(self):
+        window = self.suite.top_window()
+
+        logging.info('Start connecting to gamelet.')
+        if flags.FLAGS.enable_ui_beta:
+            connect_radio = self.find_control('RadioButton', 'ConnectToStadia')
+            connect_radio.click_input()
+
+        # Wait for the first data item in the instance list to exist
+        # We're not using find_control here because magic lookup enables us to easily wait for the existence of a row
+        window.InstanceList.click_input()
+        window.InstanceList.DataItem0.wait('exists', timeout=100)
+        instance_list = self.find_control('Table', 'InstanceList')
+        logging.info('Found %s rows in the instance list', instance_list.item_count())
+        self.expect_true(instance_list.item_count() >= 1, 'Found at least one instance')
+
+        window.InstanceList.DataItem0.double_click_input()
+        logging.info('Connecting to Instance, waiting for the process list...')
+
+        if flags.FLAGS.enable_ui_beta:
+            # In the new UI, use small waits until the process list is active, and then some more for the Overlay
```

Comment:
It's part of the tables in the new connection window, the semi-transparent black overlay indicting progress while connecting. This is not ideal, but right now there is no way for the automation to know when loading of the table is done because the overlay is not correctly exposed through the automation API (for reasons unknown).

But I think this should be looked into on C++ side and we could then get rid of this sleep. I'll create a bug for this and mention it here.

Reply:
Created b/177037834 and added a TODO referencing the bug in the Python code